### DISCUSSION TITLE
Configuring angular-auth-oidc-client with the new functional APIs

### DISCUSSION THREAD

**Discussion by giscus** [timdeschryver/timdeschryver.dev#382] [Announcements]
# Configuring angular-auth-oidc-client with the new functional APIs

Recently the angular-auth-oidc-client library (v16) introduced new APIs to configure the library in a functional way. In this blog post, I'll show you how to use them.

https://timdeschryver.dev/blog/configuring-angular-auth-oidc-client-with-the-new-functional-apis

**Comment by donahula** [comment]
Thanks! I am trying to do something very similar where I am using a provideAuth() method to do the configuration and initialization. I have no problems when I hard-code the values as shown in your example. But what if these values are themselves coming from an API call that is performed as part of the APP_INITIALIZER token? How can I dynamically pass in the config values to provideAuth()?

**Reply by timdeschryver** [reply]
I'm going to refer you to the docs, which have recently been updated to the standalone version.
You will need to use the StsConfigLoader loader.

https://angular-auth-oidc-client.com/docs/documentation/configuration#getting-static-config-from-a-service-sync
